src/box/iproto.cc

```diff
@@ -2130,6 +2130,7 @@ tx_accept_msg(struct cmsg *m)
 	rlist_add_entry(&msg->connection->tx.inprogress, msg,
 			in_inprogress);
 	msg->fiber = fiber();
+	fiber_set_system(fiber(), false);
```

Comment:
We shouldn't reset the flag here for all fibers just in case. It should be set before we start system work and unset right after system work is done. You just want to treat idle pool fibers as non-cancellable, better set/clear the flag in the fiber pool with a proper comment.

Reply:
Done. Now instead cbus message callback should set system flag for the fiber. Fibers inside pool are considered system.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/iproto.cc b/src/box/iproto.cc
index a640661e9..a51810116 100644
--- a/src/box/iproto.cc
+++ b/src/box/iproto.cc
@@ -2130,7 +2130,6 @@ tx_accept_msg(struct cmsg *m)
 	rlist_add_entry(&msg->connection->tx.inprogress, msg,
 			in_inprogress);
 	msg->fiber = fiber();
-	fiber_set_system(fiber(), false);
 	rmean_collect(msg->connection->iproto_thread->tx.rmean,
 		      REQUESTS_IN_PROGRESS, 1);
 	flightrec_write_request(msg->reqstart, msg->len);
@@ -2184,7 +2183,6 @@ tx_end_msg(struct iproto_msg *msg, struct obuf_svp *svp)
 	msg->connection->iproto_thread->tx.requests_in_progress--;
 	rlist_del(&msg->in_inprogress);
 	msg->fiber = NULL;
-	fiber_set_system(fiber(), true);
 	struct obuf *out = msg->connection->tx.p_obuf;
 	if (msg->connection->tx.p_obuf->used != svp->used)
 		/* Log response to the flight recorder. */
diff --git a/src/box/vy_scheduler.c b/src/box/vy_scheduler.c
index 208b6695d..680ab935c 100644
--- a/src/box/vy_scheduler.c
+++ b/src/box/vy_scheduler.c
@@ -890,6 +890,7 @@ vy_deferred_delete_batch_process_f(struct cmsg *cmsg)
 	struct vy_deferred_delete_batch *batch = container_of(cmsg,
 				struct vy_deferred_delete_batch, cmsg);
 	struct vy_task *task = batch->task;
+	fiber_set_system(fiber(), true);
 	/*
 	 * Wait for memory quota if necessary before starting to
 	 * process the batch (we can't yield between statements).
diff --git a/src/lib/core/fiber_pool.c b/src/lib/core/fiber_pool.c
index e15532516..e24f4f461 100644
--- a/src/lib/core/fiber_pool.c
+++ b/src/lib/core/fiber_pool.c
@@ -61,7 +61,9 @@ fiber_pool_f(va_list ap)
 			f->caller->flags |= FIBER_IS_READY;
 			assert(f->caller->caller == &cord->sched);
 		}
+		fiber_set_system(fiber(), false);
 		cmsg_deliver(msg);
+		fiber_set_system(fiber(), true);
 		fiber_check_gc();
 		/*
 		 * Normally fibers die after their function
@@ -131,7 +133,7 @@ fiber_pool_cb(ev_loop *loop, struct ev_watcher *watcher, int events)
 			f = rlist_shift_entry(&pool->idle, struct fiber, state);
 			fiber_call(f);
 		} else if (pool->size < pool->max_size) {
-			f = fiber_new_system(cord_name(cord()), fiber_pool_f);
+			f = fiber_new(cord_name(cord()), fiber_pool_f);
 			if (f == NULL) {
 				diag_log();
 				break;

```

</details>